lib/tmp.js

```diff
@@ -220,6 +220,14 @@ function _createTmpFile(options, callback) {
     fs.open(name, CREATE_FLAGS, opts.mode || FILE_MODE, function _fileCreated(err, fd) {
       if (err) return cb(err);
 
+      if (opts.discardDescriptor) {
+        return fs.close(fd, function _invokeCallback(err) {
+          cb(null, name, undefined, _prepareTmpFileRemoveCallback(name, -1, opts));
```

Comment:
I do in fact know sinon, and I use it extensively in projects I maintain where I target 100% unit test coverage.  I'm not going to add it to node-tmp, especially since its unit tests are already fragile (they leave a lot of garbage in /tmp after they're run, and are non-deterministic as to whether they can be safely run repeatedly).